lib/compat/wordpress-6.4/block-patterns.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Overrides Core's wp-includes/block-patterns.php to add new wp_patterns taxonomy for WP 6.4.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds a new taxonomy for organizing user created patterns.
+ *
+ * Note: This should be removed when the minimum required WP version is >= 6.4.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/53163
+ *
+ * @return void
+ */
+function gutenberg_register_taxonomy_patterns() {
+	$labels = array(
+		'name'          => _x( 'Pattern Categories', 'taxonomy general name' ),
+		'singular_name' => _x( 'Pattern Category', 'taxonomy singular name' ),
+	);
+	$args   = array(
+		'hierarchical'      => false,
+		'labels'            => $labels,
+		'show_ui'           => true,
+		'show_in_menu'      => false,
+		'show_in_nav_menus' => false,
+		'show_admin_column' => true,
+		'query_var'         => true,
+		'show_in_rest'      => true,
+		'_builtin'          => true,
+		'rewrite'           => array( 'slug' => 'wp_pattern_category' ),
```

Comment:
* We can set `show_ui`  to `false`; the post type has no admin menu, which will do nothing.
* REST API will mainly consume the taxonomy, so it's not publicly queriable. Related arguments (`query_var`, `rewrite`, `public`) will need to be adjusted.

That would give us pretty much the same argument shape as `wp_template_part_area` - https://github.com/WordPress/wordpress-develop/blob/8cd99074624a8c4b589f31c7b2a8092f4fe5c79e/src/wp-includes/taxonomy.php#L207-L224

Reply:
The only difference with patterns is if we set show_ui false and make it not public users won't be able to see/update the terms in the `/wp-admin/edit.php?post_type=wp_block` interface, eg.

<img width="926" alt="Screenshot 2023-08-17 at 4 48 05 PM" src="https://github.com/WordPress/gutenberg/assets/3629020/d170b91d-25f9-4697-85c0-b45791c68cbe">

Not sure how critical that is 🤔 What do you think?

Reply:
The `PostTaxonomies` component [only checks for `show_ui`](https://github.com/WordPress/gutenberg/blob/ff06828dbffc90b3d736f6cc514067cda03a2c4d/packages/editor/src/components/post-taxonomies/index.js#L27-L29), not the other arguments.

Considering that the new taxonomy does very little without remaining work from #53164. I think it's okay to hide UI.

Reply:
true, have removed the UI for now at least.

Reply:
Thank you, Glen!